hi-ml-cpath/src/health_cpath/utils/plots_utils.py

```diff
@@ -66,26 +68,42 @@ def save_pr_curve(results: ResultsType, figures_dir: Path, stage: str = '') -> N
         scores = get_list_from_results_dict(results=results, results_key=ResultsKey.PROB)
         fig, ax = plt.subplots()
         plot_pr_curve(true_labels, scores, legend_label=stage, ax=ax)
+        if stratify_metadata is not None:
+            stratified_outputs = get_stratified_outputs(true_labels=true_labels, scores=scores,
+                                                        stratify_metadata=stratify_metadata)
+            for key in stratified_outputs.keys():
+                true_stratified = stratified_outputs[key][0]
+                pred_stratified = stratified_outputs[key][1]
+                plot_pr_curve(true_stratified, pred_stratified, legend_label=f"{stage}_{key}", ax=ax)
         ax.legend()
         format_pr_or_roc_axes(plot_type='pr', ax=ax)
         save_figure(fig=fig, figpath=figures_dir / f"pr_curve_{stage}.png")
     else:
         logging.warning("The PR curve plot implementation works only for binary cases, this plot will be skipped.")
 
 
-def save_roc_curve(results: ResultsType, figures_dir: Path, stage: str = '') -> None:
+def save_roc_curve(results: ResultsType, figures_dir: Path, stage: str = '',
```

Comment:
do we have a test already for save_roc_curve, where we could also test the output if we are stratifying? It would be good to see in the tests how the plots come out with stratification.

Reply:
I have now added the test and the output, thanks